src/z3c/relationfield/event.py

```diff
@@ -73,7 +73,8 @@ def updateRelations(obj, event):
     # get errors involving buckets changing size
     rels = list(catalog.findRelations({'from_id': obj_id}))
     for rel in rels:
-        catalog.unindex(rel)
+        if hasattr(obj, rel.from_attribute):
```

Comment:
``hasattr`` has the drawback to hide any error when accessing, but I think this is fine in this context as ``from_attribute`` isn't calculated.